bionic/core/flow_execution.py

```diff
@@ -599,10 +603,32 @@ class ExecutionContext:
     flow_instance_uuid = attr.ib()
     core = attr.ib()
     task_key_logger = attr.ib()
+    # This is used for temporarily saving results for the duration of one execution.
+    # Currently we only do this for entities where both persistence and memoization are
+    # disabled.
+    temp_result_cache = attr.ib()
```

Comment:
Given that this cache lives inside `ExecutionContext`, I think it's clear that this is a temporary cache. IMO, you can drop `temp`.

Reply:
I guess it doesn't seem super obvious to me, especially since the other memory cache is handled in a different way. For example, in the line

    self.context.temp_result_cache.save(result)

I think the `temp_` makes it clearer that this is distinct from the main memory cache.